scripts/test-v1-v2-rollover.ts

```diff
@@ -0,0 +1,149 @@
+    const VAULT_FACTORY_ADDRESS = "0x6e9B4c2f6Bd57b7b924d29b5dcfCa1273Ecc94A2";
+    const ADDRESSES_PROVIDER_ADDRESS = "0xB53C1a33016B2DC2fF3653530bfF1848a515c8c5";
+    const BORROWER_NOTE_ADDRESS = "0xc3231258D6Ed397Dce7a52a27f816c8f41d22151";
+
+    const [newLender] = await hre.ethers.getSigners();
+
+    const LOAN_ID = 29;
+    const NONCE = 2;
+    const repayAmount = ethers.utils.parseUnits("129687.5", 6);
+
+    await hre.network.provider.request({
+        method: "hardhat_impersonateAccount",
+        params: [WHALE],
+    });
+
+    await hre.network.provider.request({
+        method: "hardhat_impersonateAccount",
+        params: [BORROWER],
+    });
+
+    await hre.network.provider.request({
+        method: "hardhat_impersonateAccount",
+        params: [LENDER],
+    });
+
+    const borrower = await hre.ethers.getSigner(BORROWER);
+    const lender = await hre.ethers.getSigner(LENDER);
+    const whale = await hre.ethers.getSigner(WHALE);
+
+    const erc20Factory = await ethers.getContractFactory("ERC20");
+    const usdc = <ERC20>await erc20Factory.attach(USDC_ADDRESS);
+
+    const erc721Factory = await ethers.getContractFactory("ERC721");
+    const note = <PromissoryNote>await erc721Factory.attach(BORROWER_NOTE_ADDRESS);
+
+    console.log("Deploying rollover...");
+
+    const factory = await ethers.getContractFactory("FlashRolloverV1toV2")
+    const flashRollover = <FlashRolloverV1toV2>await factory.deploy(ADDRESSES_PROVIDER_ADDRESS);
+
+    console.log("Doing approvals...");
+
+    await whale.sendTransaction({ to: borrower.address, value: ethers.utils.parseEther("100") });
+    await whale.sendTransaction({ to: lender.address, value: ethers.utils.parseEther("100") });
+    await whale.sendTransaction({ to: newLender.address, value: ethers.utils.parseEther("100") });
+    await usdc.connect(whale).transfer(newLender.address, ethers.utils.parseUnits("1000000", 6))
+    await usdc.connect(whale).transfer(borrower.address, ethers.utils.parseUnits("100000", 6))
+
+    // Lender approves USDC
+    await usdc.connect(newLender).approve(OC_ADDRESS, ethers.utils.parseUnits("100000000000", 6));
+
+    // Borrower approves USDC and borrower note
+    await usdc.connect(borrower).approve(flashRollover.address, ethers.utils.parseUnits("100000000000", 6));
+    await note.connect(borrower).approve(flashRollover.address, LOAN_ID);
+
+    console.log("Creating a vault...");
+
+    // Lender creates vault
+    const vfFactory = await ethers.getContractFactory("VaultFactory");
+    const vaultFactory = <VaultFactory>await vfFactory.attach(VAULT_FACTORY_ADDRESS);
+
+    const initTx = await vaultFactory.connect(borrower).initializeBundle(borrower.address);
+    const initReceipt = await initTx.wait();
+
+    const createdEvent = initReceipt.events?.find(e => e.event === "VaultCreated");
+    const vault = createdEvent?.args?.[0];
+
+    console.log(`Approving vault ${vault}...`);
+
+    // Lender approves vault
```

Comment:
nit: shouldn't this read `// Borrower approves vault` ?

Reply:
Fixed